lib/top_hangman/io.rb

```diff
@@ -0,0 +1,55 @@
+module TopHangman
+  class IO
+    class << self
+      def execute(game, renderer)
+        game.start
+
+        while game.state == Game::RUNNING
+          renderer.show_header(errors_count: game.current_round.errors_count)
+
+          execute_round_loop(renderer, game) while game.current_round.state == TopHangman::Round::RUNNING
+
+          show_ending_round_message(renderer, game)
+          ask_for_new_round(renderer, game)
+        end
+      end
+
+      def execute_round_loop(renderer, game)
+        renderer.ask_for_guess
+        guess_letter = gets.chomp
+        game.play_round(guess_letter)
+        renderer.show_progress(
+          errors_count: game.current_round.errors_count,
+          guess_history: game.current_round.guess_history,
+          word: game.current_round.word
+        )
+      end
+
+      def show_ending_round_message(renderer, game)
+        case game.current_round.result
+        when TopHangman::Round::WON
+          renderer.winning_message(
+            errors_count: game.current_round.errors_count,
+            word: game.current_round.word
+          )
+        when TopHangman::Round::LOST
+          renderer.losing_message(
+            errors_count: game.current_round.errors_count,
+            word: game.current_round.word
+          )
+        end
+      end
+
+      def ask_for_new_round(renderer, game)
+        gets
```

Comment:
what is this `gets` for?

Reply:
<img width="342" alt="image" src="https://user-images.githubusercontent.com/26731448/194919555-275267af-fea2-41fa-b7e2-4af3900ed761.png">

for this

Reply:
oh got it. I'd leave it in a separate method instead of including it but it's ok